server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -1700,6 +1700,9 @@ public static boolean canMatchSearchAfter(
             && primarySortField.missing() == null
             && Objects.equals(trackTotalHitsUpto, SearchContext.TRACK_TOTAL_HITS_DISABLED)) {
             final Object searchAfterPrimary = searchAfter.fields[0];
+            if (searchAfterPrimary == null) {
```

Comment:
@gaobinlong could you please explain this change? thank you

Reply:
Oh, it shouldn't be here, it is some verification code for addressing another issue, I've removed this change, thank you!